nx/lib/nx/random.ex

```diff
@@ -80,22 +80,33 @@ defmodule Nx.Random do
         [0, 12]
       >
 
-  A single key effectively consists of 64 bits, so all possible values
-  of a 64-bit integer result in a different key. However, when passing
-  an integer literal, Nx implicitly assumes its type to be `:s32`,
-  which would result in an overflow for large integers. Therefore,
-  when dealing with large seeds, make sure to explicitly use a 64 bit
-  type:
-
-      iex> Nx.Random.key(Nx.u64(999999999999))
+      iex> Nx.Random.key(999999999999)
       #Nx.Tensor<
         u32[2]
         [232, 3567587327]
       >
   """
-  defn key(seed) do
-    seed = Nx.as_type(seed, :u64)
+  deftransform key(seed) do
+    seed =
+      case seed do
+        seed when is_integer(seed) ->
+          Nx.u64(seed)
 
+        %Nx.Tensor{} = seed when seed.type == {:u, 64} ->
+          seed
+
+        %Nx.Tensor{} = seed when seed.type == {:s, 64} ->
+          Nx.bitcast(seed, {:u, 64})
+
+        other ->
+          raise ArgumentError,
+                "expected seed to be an integer, u64 tensor or s64 tensor, got: #{inspect(other)}"
+      end
+
+    key_n(seed)
+  end
+
+  defnp key_n(seed) do
```

Comment:
```suggestion
  deftransform key(seed) do
    if is_integer(seed) do    
      key_n(Nx.u64(seed))
    else
      key_n(seed)
    end
  end

  defnp key_n(seed) do
    seed = Nx.as_type(seed, :u64)
```

Reply:
I think requiring :u64 is fine. The point about the validation is to avoid users accidentally lowering the seed space by passing s32 tensors inside defn.

Reply:
I see though that you want to avoid an integer passed as argument, which will be downcast to s32, from becoming a key. Yeah, ignore me.

Reply:
FWIW jax does what you said :) ([ref](https://github.com/google/jax/blob/f90558d317936281d3edae74e05027939a106841/jax/_src/prng.py#L529-L535))

Reply:
I ran Axon suite and it all worked. So I think we will be fine.

Reply:
I think this approach is alot simpler too